packages/mdc-menu/foundation.ts

```diff
@@ -131,19 +131,33 @@ export class MDCMenuFoundation extends MDCFoundation<MDCMenuAdapter> {
     this.defaultFocusState_ = focusState;
   }
 
+  /**
+   * Selects the list item at `index` witnin the specified selection group.
+   * @param index Index of list item within the selection group.
+   * @param selectionGroupIndex Index of selection group within the menu.
+   */
+  setSelectedIndex(index: number, selectionGroupIndex: number) {
+    const selectionGroups = this.adapter_.getMenuSelectionGroups();
+    this.handleSelectionGroup_(selectionGroups[selectionGroupIndex], index);
+  }
+
   /**
    * Handles toggling the selected classes in a selection group when a selection is made.
    */
-  private handleSelectionGroup_(selectionGroup: Element, index: number) {
+  private handleSelectionGroup_(selectionGroup: Element | null, index: number) {
+    if (!selectionGroup) {
+      return;
+    }
     // De-select the previous selection in this group.
     const selectedIndex = this.adapter_.getSelectedElementIndex(selectionGroup);
     if (selectedIndex >= 0) {
       this.adapter_.removeAttributeFromElementAtIndex(selectedIndex, strings.ARIA_SELECTED_ATTR);
       this.adapter_.removeClassFromElementAtIndex(selectedIndex, cssClasses.MENU_SELECTED_LIST_ITEM);
     }
     // Select the new list item in this group.
-    this.adapter_.addClassToElementAtIndex(index, cssClasses.MENU_SELECTED_LIST_ITEM);
-    this.adapter_.addAttributeToElementAtIndex(index, strings.ARIA_SELECTED_ATTR, 'true');
+    const normalizedIndex = this.adapter_.getListItemIndexOfSelectionGroup(index, selectionGroup);
```

Comment:
Alternatively the `setSelectedIndex` API can accept the root list indexing. This approach simplifies our implementation and also avoids confusion using two separate indexing (ie., indexing for main list and indexing for each selection group). It makes the component consumer easier to just use list indexing everywhere. 

Majority of the time that I've seen Menu has single selection group. `setSelectedIndex` will work even though there are multiple selection groups as expected.


API signature:

```ts
setSelectedIndex(index: number)
```

Example:

```ts
setSelectedIndex(5) // index '0' starts from first list item always.
```

WDYT?